sdk/keyvault/azure-security-keyvault-secrets/src/main/java/com/azure/security/keyvault/secrets/models/KeyVaultSecret.java

```diff
@@ -98,41 +101,53 @@ public KeyVaultSecret setProperties(SecretProperties properties) {
         return this;
     }
 
-    @JsonProperty(value = "id")
-    private void unpackId(String id) {
-        properties.unpackId(id);
+    @Override
+    public JsonWriter toJson(JsonWriter jsonWriter) throws IOException {
+        return jsonWriter.writeStartObject()
+            .writeStringField("value", value)
+            .writeEndObject();
     }
 
     /**
-     * Unpacks the attributes json response and updates the variables in the Secret Attributes object.
-     * Uses Lazy Update to set values for variables id, tags, contentType, managed and keyId as these variables are
-     * part of main json body and not attributes json body when the secret response comes from list Secrets operations.
-     * @param attributes The key value mapping of the Secret attributes
+     * Reads an instance of {@link KeyVaultSecret} from the JsonReader.
+     *
+     * @param jsonReader The JsonReader being read.
+     * @return An instance of {@link KeyVaultSecret} if the JsonReader was pointing to an instance of it, or null if it
+     * was pointing to JSON null.
+     * @throws IOException If an error occurs while reading the {@link KeyVaultSecret}.
      */
-    @JsonProperty("attributes")
-    @SuppressWarnings("unchecked")
-    private void unpackAttributes(Map<String, Object> attributes) {
-        properties.unpackAttributes(attributes);
-    }
-
-    @JsonProperty("managed")
-    private void unpackManaged(Boolean managed) {
-        properties.managed = managed;
-    }
-
-    @JsonProperty("kid")
-    private void unpackKid(String kid) {
-        properties.keyId = kid;
-    }
-
-    @JsonProperty("contentType")
-    private void unpackContentType(String contentType) {
-        properties.contentType = contentType;
-    }
-
-    @JsonProperty("tags")
-    private void unpackTags(Map<String, String> tags) {
-        properties.tags = tags;
+    public static KeyVaultSecret fromJson(JsonReader jsonReader) throws IOException {
```

Comment:
Do we need to make these to/fromJson methods public? Is this something we do for all our models generated with AutoRest?

Reply:
We do need to make these methods public and this is part of the migration away from Jackson Databind as the serialization mechanism for our models. This allows for external usage of models in serialization in either Jackson or GSON by using the interopability layers provided by `azure-core-serializer-json-jackson` and `azure-core-serializer-json-gson` as they include databind-style modules that will hook into either a Jackson ObjectMapper or GSON Gson.